app/Http/Controllers/UserController.php

```diff
@@ -2,9 +2,83 @@
 
 namespace App\Http\Controllers;
 
-class UserController
+use App\Http\Resources\Authentication\UserResource;
+use Exception;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\URL;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Http;
+use App\Http\Resources\ErrorResource;
+use App\Http\Resources\SuccessResource;
+use Illuminate\Support\Facades\Validator;
+
+class UserController extends Controller
 {
 
+    public function clientToken(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
+            'client_id' => 'required',
+            'client_secret' => 'required',
+        ]);
+
+        if ($validator->fails()) {
```

Comment:
@kshitij1326  Please move validator in a new PHP file. It is repeating quite often.

Reply:
@kshitij1326 I implemented a new Validator abstract class and rules. It will reduce the load from the controllers and the controllers will be thin. 
Please take a look, if you find it bit weird then please do let me know. 

I tried not to the change the logic of validation that you implemented. I just refactor the code as of now